depends/packages/qt.mk

```diff
@@ -140,7 +140,11 @@ $(package)_config_opts_linux += -no-feature-vulkan
 $(package)_config_opts_linux += -dbus-runtime
 $(package)_config_opts_arm_linux += -platform linux-g++ -xplatform bitcoin-linux-g++
 $(package)_config_opts_i686_linux  = -xplatform linux-g++-32
+ifneq (,$(findstring clang++ -stdlib=libc++,$($(1)_cxx)))
```

Comment:
As mentioned in the comment from https://github.com/bitcoin/bitcoin/pull/22815#discussion_r697683792:

> Looking only at the argument (which should directly follow the compiler name) looks kind of fragile and will not catch every case where someone links against libc++.

To expand on that, as-is this also wont work when you're using a versioned clang, i.e `clang++-12`, which we often do in our CIs.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/22814#issuecomment-928891018).